packages/core-backend/src/AccountActivityService.test.ts

```diff
@@ -0,0 +1,2936 @@
+/* eslint-disable jest/no-conditional-in-test */
+import { Messenger } from '@metamask/base-controller';
+import type { InternalAccount } from '@metamask/keyring-internal-api';
+import type { Hex } from '@metamask/utils';
+
+import {
+  AccountActivityService,
+  type AccountActivityServiceMessenger,
+  type AccountSubscription,
+  ACCOUNT_ACTIVITY_SERVICE_ALLOWED_ACTIONS,
+  ACCOUNT_ACTIVITY_SERVICE_ALLOWED_EVENTS,
+} from './AccountActivityService';
+import type {
+  WebSocketConnectionInfo,
+  ServerNotificationMessage,
+} from './BackendWebSocketService';
+import { WebSocketState } from './BackendWebSocketService';
+import type { AccountActivityMessage } from './types';
+
+// Mock global fetch for API testing
+const mockFetch = jest.fn();
+global.fetch = mockFetch;
+
+// Test helper constants - using string literals to avoid import errors
+enum ChainId {
+  mainnet = '0x1',
+  sepolia = '0xaa36a7',
+}
+
+// Mock function to create test accounts
+const createMockInternalAccount = (options: {
+  address: string;
+}): InternalAccount => ({
+  address: options.address.toLowerCase() as Hex,
+  id: `test-account-${options.address.slice(-6)}`,
+  metadata: {
+    name: 'Test Account',
+    importTime: Date.now(),
+    keyring: {
+      type: 'HD Key Tree',
+    },
+  },
+  options: {},
+  methods: [],
+  type: 'eip155:eoa',
+  scopes: ['eip155:1'], // Required scopes property
+});
+
+/**
+ * Creates a real messenger with registered mock actions for testing
+ * Each call creates a completely independent messenger to ensure test isolation
+ *
+ * @returns Object containing the messenger and mock action functions
+ */
+const createMockMessenger = () => {
+  // Use any types for the root messenger to avoid complex type constraints in tests
+  // Create a unique root messenger for each test
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const rootMessenger = new Messenger<any, any>();
+  const messenger = rootMessenger.getRestricted({
+    name: 'AccountActivityService',
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    allowedActions: [...ACCOUNT_ACTIVITY_SERVICE_ALLOWED_ACTIONS] as any,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    allowedEvents: [...ACCOUNT_ACTIVITY_SERVICE_ALLOWED_EVENTS] as any,
+  }) as unknown as AccountActivityServiceMessenger;
```

Comment:
nit - `any` type cleanup:
```suggestion
  const rootMessenger = new Messenger<
    AccountActivityServiceAllowedActions,
    AccountActivityServiceAllowedEvents
  >();
  const messenger: AccountActivityServiceMessenger =
    rootMessenger.getRestricted({
      name: 'AccountActivityService',
      allowedActions: [...ACCOUNT_ACTIVITY_SERVICE_ALLOWED_ACTIONS],
      allowedEvents: [...ACCOUNT_ACTIVITY_SERVICE_ALLOWED_EVENTS],
    });
```